### PR TITLE
Upgrade to springboot 2.5.4

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**

### CODE REVIEW

File: oauth2/pom.xml [hunk 1]
```diff
@@ -39,6 +39,12 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
```
Comment (arvindkrishnakumar-okta) [inline oauth2/pom.xml:42]: Context: https://github.com/okta/okta-spring-boot/pull/331#issuecomment-918758294

Comment (mraible) [reply oauth2/pom.xml:42]: A better import might be the Validation API:

```xml
<dependency>
    <groupId>javax.validation</groupId>
    <artifactId>validation-api</artifactId>
</dependency>
```

Then, change the import to:

```java
import javax.validation.constraints.NotNull;
```

WDYT @bdemers?

I tried this and all tests pass.

**Comment by mraible** [comment]
NOTE: It seems that `OktaJwtAuthenticationConverter#extractAuthorities()` overrides a deprecated method.

File: oauth2/pom.xml (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply oauth2/pom.xml:42]: Sounds good to me, updated PR.

Comment (bdemers) [reply oauth2/pom.xml:42]: 💯 The version will be managed automatically too.

**Review by mraible — APPROVED** [review]

**Review by bdemers — APPROVED** [review]

**Comment by bdemers** [comment]
There were two test deps out of date rest-assured and TestNG,
The rest-assured bump is here: #342 

Updating TestNG will require updating Surefire and Failsafe in the parent pom (to 3.x), but that shouldn't block a release

**Pull request merged**